src/main/java/org/asciidoctor/asciidoclet/AsciidoctorConverter.java

```diff
@@ -181,7 +181,21 @@ private String convert(String input, boolean inline) {
         if (input.trim().isEmpty()) {
             return "";
         }
-        options.setDocType(inline ? INLINE_DOCTYPE : "");
+        // Setting doctype to null results in an NPE from asciidoctor.
+        // the default value from the command line is "article".
+        // https://docs.asciidoctor.org/asciidoctor/latest/cli/man1/asciidoctor/#options
+        // In general, in order to respect original doctype, we should do the following.
```

Comment:
Your guess is absolutely right.
The test fails because `options` is not restored, so when the converter converts the comment text it sets docType=inline, then when it converts the param, it checks if docType is already set and it finds it's inline, so it applies that.
A quick solution would be to remove docType after running the conversion

```
      String converted = asciidoctor.convert(cleanJavadocInput(input), options);
      options.setDocType(null);
      return converted;
```

But a better thing would be to make options immutable, simply create a new instance on every conversion instead of re-using it, but that requires more time. I leave that to you.

Reply:
I tried the change you suggested and it made asciidoctor throw NPE from a different line.
```
java.lang.NullPointerException: Cannot invoke "Object.toString()" because the return value of "java.util.Map.get(Object)" is null

	at asciidoctorj@2.5.12/org.asciidoctor.jruby.internal.AsciidoctorUtils.getOptions(AsciidoctorUtils.java:107)
	at asciidoctorj@2.5.12/org.asciidoctor.jruby.internal.AsciidoctorUtils.toAsciidoctorCommand(AsciidoctorUtils.java:45)
	at asciidoctorj@2.5.12/org.asciidoctor.jruby.internal.JRubyAsciidoctor.convert(JRubyAsciidoctor.java:300)
	at asciidoctorj@2.5.12/org.asciidoctor.jruby.internal.JRubyAsciidoctor.convert(JRubyAsciidoctor.java:346)
	at asciidoctorj@2.5.12/org.asciidoctor.jruby.internal.JRubyAsciidoctor.convert(JRubyAsciidoctor.java:341)
```

Also, the other option would result in a wide-spread change.
So, how about keeping it as is for now and fling a ticket to address the issue in more decent way?

Reply:
This works
```
options.setDocType("");
```

But given the amount of workarounds already in place, I agree it is better to treat separately and do it right :point_right: https://github.com/asciidoctor/asciidoclet/issues/125